src/Scheduler/StatefulScheduledCommandsResolver.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use DateTime;
+use Tempest\Console\ConsoleCommand;
+
+final class StatefulScheduledCommandsResolver implements ScheduledCommandsResolver
+{
+    public const string CACHE_PATH = __DIR__ . '/last-schedule-run.cache.php';
+
+    public function __construct(
+        private SchedulerConfig $config,
+    ) {
+    }
+
+    public function resolve(DateTime $date): array
```

Comment:
This should be a better name. I actually think all of this code should be moved to within `Scheduler`, until we decide to add multiple scheduling strategies — which is out of scope for now